libraries/AC_PrecLand/AC_PrecLand.cpp

```diff
@@ -693,13 +695,44 @@ bool AC_PrecLand::construct_pos_meas_using_rangefinder(float rangefinder_alt_m,
     return false;
 }
 
+// Set the active inertial frame, based on either sensor timestamp (higher priority) or PLND_LAG parameter
+// Return true if sensor does not send timestamp and front of buffer is used, or if sensor timestamp is successfully synced
+// Return false otherwise, so invalid sync is not propagated
+bool AC_PrecLand::sync_frames(const uint32_t los_meas_time_ms)
+{
+    // Retrieve timestamp from backend measurement.  If not zero, attempt to correlate with inertial history.
+    uint64_t nearest_timestamp = 0;
+    uint16_t nearest_index = 0;
+    const uint64_t los_meas_time_us = los_meas_time_ms * 1000;
+    if (los_meas_time_us != 0) {
+        // Search through inertial history for the nearest frame to the measurement timestamp
+        for (uint8_t i=0; i<_inertial_history->available(); i++) {
+            uint64_t delta_usec = labs((*_inertial_history)[i]->time_usec - los_meas_time_us);
```

Comment:
is there a time wrap issue here?

Reply:
los_meas_time_ms will wrap after 71 days, and time_usec does not wrap, so there is a wrap issue

Reply:
I solved that now and unit tested it with the script below.